src/localkiwixserver.cpp

```diff
@@ -0,0 +1,71 @@
+#include "localkiwixserver.h"
+#include "ui_localkiwixserver.h"
+#include "kiwixapp.h"
+#include <QDesktopServices>
+#include <QMessageBox>
+
+LocalKiwixServer::LocalKiwixServer(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::LocalKiwixServer)
+{
+    ui->setupUi(this);
+
+    QFile styleFile(":/css/localServer.css");
+    styleFile.open(QIODevice::ReadOnly);
+    auto byteContent = styleFile.readAll();
+    styleFile.close();
+    QString style(byteContent);
+    setStyleSheet(style);
+
+    connect(ui->KiwixServerButton, SIGNAL(clicked()), this, SLOT(runOrStopServer()));
+    connect(ui->OpenInBrowserButton, SIGNAL(clicked()), this, SLOT(openInBrowser()));
+    const QHostAddress &localhost = QHostAddress(QHostAddress::LocalHost);
+    for (const QHostAddress &address: QNetworkInterface::allAddresses()) {
+        if (address.protocol() == QAbstractSocket::IPv4Protocol && address != localhost) {
+            m_ipAddress = address.toString();
+            ui->IpAddress->setText(m_ipAddress);
+            break;
+        }
+    }
+    ui->OpenInBrowserButton->setVisible(false);
+    ui->IpAddress->setVisible(false);
+}
+
+LocalKiwixServer::~LocalKiwixServer()
+{
+    delete ui;
+}
+
+void LocalKiwixServer::openInBrowser()
+{
+    QUrl url;
+    url.setScheme("http");
+    url.setHost(m_ipAddress);
+    url.setPort(8181);
+    QDesktopServices::openUrl(url);
+}
+
+void LocalKiwixServer::runOrStopServer()
+{
+    if (KiwixApp::instance()->isLocalServerRunning()) {
+        QMessageBox messageBox;
+        messageBox.critical(0,"Error","An error has occured !");
+        return;
+    }
+    m_active = !m_active;
+    if (m_active) {
+        KiwixApp::instance()->runServer();
```

Comment:
We should check here that we have correctly launch the server.
If kiwix-serve is not present, the serve is not launch (normal) but everything seems to work correctly. The "open in browser" just open the browser using a address where nothing is served.